internal/cmd/egctl/status.go

```diff
@@ -0,0 +1,324 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package egctl
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"reflect"
+	"strconv"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	gwv1 "sigs.k8s.io/gateway-api/apis/v1"
+	gwv1a2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+
+	egv1a1 "github.com/envoyproxy/gateway/api/v1alpha1"
+)
+
+func newStatusCommand() *cobra.Command {
+	var (
+		quiet, verbose, allNamespaces bool
+		resourceType, namespace       string
+	)
+
+	statusCommand := &cobra.Command{
+		Use:   "status",
+		Short: "Show the summary of the status of resources in Envoy Gateway",
+		Example: `  # Show the status of gatewayclass resources under default namespace.
+  egctl x status gatewayclass
+
+  # Show the status of gateway resources with less information under default namespace.
+  egctl x status gateway -q
+
+  # Show the status of gateway resources with details under default namespace.
+  egctl x status gateway -v
+
+  # Show the status of httproutes resources with details under a specific namespace.
+  egctl x status httproutes -v -n foobar
+
+  # Show the status of httproutes resources under all namespaces.
+  egctl x status httproutes -A
+	`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+
+			table := newStatusTableWriter(os.Stdout)
+			k8sClient, err := newK8sClient()
+			if err != nil {
+				return err
+			}
+
+			switch {
+			case len(args) == 1:
+				resourceType = args[0]
+			case len(args) > 1:
+				return fmt.Errorf("unknown args: %s", strings.Join(args[1:], ","))
+			default:
+				return fmt.Errorf("invalid args: must specific a resources type")
+			}
+
+			return runStatus(ctx, k8sClient, table, resourceType, namespace, quiet, verbose, allNamespaces)
+		},
+	}
+
+	statusCommand.PersistentFlags().BoolVarP(&quiet, "quiet", "q", false, "Show the status of resources only")
+	statusCommand.PersistentFlags().BoolVarP(&verbose, "verbose", "v", false, "Show the status of resources with details")
+	statusCommand.PersistentFlags().BoolVarP(&allNamespaces, "all-namespaces", "A", false, "Get resources from all namespaces")
+	statusCommand.PersistentFlags().StringVarP(&namespace, "namespace", "n", "default", "Specific a namespace to get resources")
+
+	return statusCommand
+}
+
+func newStatusTableWriter(out io.Writer) *tabwriter.Writer {
+	return tabwriter.NewWriter(out, 10, 0, 3, ' ', 0)
+}
+
+func runStatus(ctx context.Context, cli client.Client, table *tabwriter.Writer, resourceType, namespace string, quiet, verbose, allNamespaces bool) error {
+	var resourcesList client.ObjectList
+
+	if allNamespaces {
+		namespace = ""
+	}
+
+	switch strings.ToLower(resourceType) {
+	case "gc", "gcs", "gatewayclass", "gatewayclasses":
```

Comment:
lets only use singular terms similar to `kubectl get`

Reply:
lets do this for all APIs

Reply:
done